Assets/Scripts3D/Weapons/Weapon.cs

```diff
@@ -43,7 +72,39 @@ private void Shoot()
         {
             return;
         }
+    }
+
+    void RotationSway()
+    {
+        float mouseX = Input.GetAxisRaw("Mouse X") * _swayMultiplier;
+        float mouseY = Input.GetAxisRaw("Mouse Y") * _swayMultiplier;
+
+        Quaternion rotationX = Quaternion.AngleAxis(-mouseY, Vector3.right);
+        Quaternion rotationY = Quaternion.AngleAxis(mouseX, Vector3.up);
 
+        Quaternion targetRotation = rotationX * rotationY;
+
+        transform.localRotation = Quaternion.Slerp(transform.localRotation, targetRotation, _smooth * Time.deltaTime);
+    }
 
+    void HeadBobController()
```

Comment:
Same here. This sounds like a constructor for a Controller. 

Every method should clearly describe what it DOES, ie with a verb. Update, Apply or whatever that might be.